vectorstores/cloudsql/vectorstore.go

```diff
@@ -51,6 +55,87 @@ func NewVectorStore(ctx context.Context, engine cloudsqlutil.PostgresEngine, emb
 	return vs, nil
 }
 
+// SimilaritySearch performs a similarity search on the database using the
+// query vector.
+func (vs *VectorStore) SimilaritySearch(ctx context.Context, query string, _ int, options ...vectorstores.Option) ([]schema.Document, error) {
+	opts, err := applyOpts(options...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to apply vector store options: %w", err)
+	}
+	var documents []schema.Document
+	embedding, err := vs.embedder.EmbedQuery(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("failed embed query: %w", err)
+	}
+	operator := vs.distanceStrategy.operator()
+	searchFunction := vs.distanceStrategy.similaritySearchFunction()
+
+	columns := append(vs.metadataColumns, vs.contentColumn)
+	if vs.metadataJsonColumn != "" {
+		columns = append(columns, vs.metadataJsonColumn)
+	}
+	columnNames := strings.Join(columns, `, `)
+	whereClause := ""
+	if opts.Filters != nil {
+		whereClause = fmt.Sprintf("WHERE %s", opts.Filters)
+	}
+	stmt := fmt.Sprintf(`
+        SELECT %s, %s(%s, '%s') AS distance FROM "%s"."%s" %s ORDER BY %s %s '%s' LIMIT $1::int;`,
+		columnNames, searchFunction, vs.embeddingColumn, vectorToString(embedding), vs.schemaName, vs.tableName, whereClause, vs.embeddingColumn, operator, vectorToString(embedding))
```

Comment:
Do we need to use vectorToString? Can we just the pgx stuff here: https://github.com/pgvector/pgvector-go?tab=readme-ov-file#pgx

Reply:
done!